tensorflow/core/grappler/optimizers/auto_mixed_precision_test.cc

```diff
@@ -104,7 +102,7 @@ class AutoMixedPrecisionTest : public GrapplerTest {
     int num_gpus = GetNumAvailableGPUs();
     // If GPUs are available, require that they all satisfy the min arch.
     gpu_available_ = (num_gpus > 0);
-#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
+#if GOOGLE_CUDA 
```

Comment:
add a `#elif TENSORFLOW_USE_ROCM` here to unconditionally set `gpu_available_` to `false`.

That way you do not need to worry about modifying `GetNumAvailableGPUs` and this will still work as it will take the `else` block at line 111.

Reply:
Wouldn't we not  be actually testing builds with GPUs > gfx906 then?

Reply:
This setup ensures that the AMP optimizer code gets enabled (without which all unit tests should fail).  That is the reason why we have code to create a dummy "gfx906" device, for the case where we do not have any GPUs that support efficient FP16 compute. Since that route will work even when we do have eligible GPUs, we can choose to take that route unconditionally...does that make sense?

Reply:
I understand.